modules/partition-replicator/src/test/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManagerTest.java

```diff
@@ -324,4 +378,55 @@ void producesEventsOnPartitionDestroy() {
         assertThat(beforeReplicaDestroyedFuture.thenApply(LocalPartitionReplicaEventParameters::zonePartitionId), willBe(zonePartitionId));
         assertThat(afterReplicaDestroyedFuture.thenApply(LocalPartitionReplicaEventParameters::zonePartitionId), willBe(zonePartitionId));
     }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenAllComponentsAreStoppedFine() throws Exception {
+        assertDoesNotThrow(() -> partitionReplicaLifecycleManager.beforeNodeStop());
+
+        assertThat(partitionReplicaLifecycleManager.stopAsync(), willCompleteSuccessfully());
+
+        verify(replicaManager, times(CatalogUtils.DEFAULT_PARTITION_COUNT)).stopReplica(any());
+    }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenReplicasAreStoppedExceptionally() throws Exception {
+        when(replicaManager.stopReplica(any())).thenThrow(new NodeStoppingException());
+
+        assertDoesNotThrow(() -> partitionReplicaLifecycleManager.beforeNodeStop());
+
+        assertThat(partitionReplicaLifecycleManager.stopAsync(), willCompleteSuccessfully());
+
+        // One more because of exceptional mock above counts.
+        verify(replicaManager, times(CatalogUtils.DEFAULT_PARTITION_COUNT + 1)).stopReplica(any());
+
+        // Do reset for correct replica manager stop on tear down.
+        reset(replicaManager);
+    }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenTxStatePartitionStoragesAreStoppedExceptionally() throws Exception {
```

Comment:
Can you please explain what is the test scenario here?

Reply:
If we are simply testing that exceptions are propagated to `stopAsync` then why do we check anything else, like that the storages have been stopped?

Reply:
Because not every exception on stop process breaks `stopAsync`: broken `stopReplica` works fine and doesn't propagate the exception